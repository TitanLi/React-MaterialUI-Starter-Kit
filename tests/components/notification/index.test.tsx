/*eslint no-console: ["off", { allow: ["warn", "error"] }] */

import React from "react";
import ReactTestUtils from "react-dom/test-utils";
import { mount } from "enzyme";

import Notification from "../../../src/components/Notification";
import Type from "../../../src/components/Notification/NotificationType";

const render = ({ callBack, ...rest } = {}) => {
  if (!callBack) callBack = () => true;

  const item = mount(<Notification {...rest} onClose={callBack} />);

  if (item.mount) item.mount();
  return item;
};

describe(`Render ${Notification.displayName} tests`, () => {
  test("Render 3 items", () => {
    const item = render({
      dataSource: [
        {
          type: Type.CONFIRM,
          message: "Hello"
        },
        {
          type: Type.INFO,
          message: "Hello"
        },
        {
          type: Type.DANGER,
          message: "Hello"
        }
      ]
    });

    expect(item).toMatchSnapshot();
    expect(item.find("svg").length).toBe(6);
  });

  test("Error when no DataSource", () => {
    const original = console.error;
    console.error = jest.fn();
    render();

    expect(console.error).toHaveBeenCalled();
    console.error = original;
  });

  test("Error when DataSource is wrong type", () => {
    const original = console.error;
    console.error = jest.fn();
    render({ dataSource: [{}, {}, {}] });

    expect(console.error).toHaveBeenCalled();
    console.error = original;
  });

  test("onClose called", () => {
    const item = render({
      dataSource: [
        {
          type: Type.CONFIRM,
          message: "Hello"
        },
        {
          type: Type.INFO,
          message: "Hello"
        },
        {
          type: Type.DANGER,
          message: "Hello",
          onClose: () => true
        }
      ],
      displayIn: 2,
      subsequentDelay: 1
    });

    const p = new Promise((r, j) => {
      item.setProps({ onClose: () => r(true) });
      setTimeout(() => j(false), 4000);
    });

    return expect(p).resolves.toBe(true);
  });
});
