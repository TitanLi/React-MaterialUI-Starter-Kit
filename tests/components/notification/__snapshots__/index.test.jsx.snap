// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render WithStyles(NotificationCenter) tests Render 3 items 1`] = `
Array [
  Array [
    <button
      aria-describedby={null}
      className="MuiButtonBase-root-58 MuiButton-root-32 MuiButton-text-34 MuiButton-flat-37 RegularButton-button-15"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      title={null}
      type="button"
    >
      <span
        className="MuiButton-label-33"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-61"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-90"
      />
    </button>,
    "",
  ],
  "",
  "",
]
`;
